packages/react-native-web/src/exports/Image/index.js

```diff
@@ -332,14 +343,76 @@ const Image: React.AbstractComponent<
   );
 });
 
-Image.displayName = 'Image';
+BaseImage.displayName = 'Image';
+
+/**
+ * This component handles specifically loading an image source with headers
+ * default source is never loaded using headers
+ */
+const ImageWithHeaders: ImageComponent = React.forwardRef((props, ref) => {
+  // $FlowIgnore: This component would only be rendered when `source` matches `ImageSource`
+  const nextSource: ImageSource = props.source;
```

Comment:
Perhaps there's a way to declare `ImageWithHeaders` as an `Image` component where the `source` prop is exactly `type ImageSource`, but I don't know exactly how 

I guess I need to declare `ImageWithHeaderProp` type that spreads the default Image prop type but changes the `source`. Should I do that?